modules/ROOT/pages/as2/as2-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,249 @@
+
+* Users can reuse their HTTP listener configuration. Doing so means several changes for you and your trading partners:
+
+** Changing ports for an AS2-receive endpoint/port with the new HTTP listener's endpoint/port configuration
+** Making your trading partners aware of the new URL/port configuration that corresponds to the HTTP listener you are reusing
+** Changes to the load-balancer, if used
+
+* Starting with this version of the AS2 connector, a broader set of ciphers are supported: `AES128_CBC`,`AES192_CBC`,`AES256_CBC`.
+Ciphers such as: `AES128`, `AES192`, `AES256` that were used in earlier versions of the AS2 connector now map to this expanded set.
+
+* The namespace has changed from `as2` to `as2-mule4`. You must delete the old namespace before adding the new one.
+
+* The single global configuration is now divided into separate operation/source global configurations for a "receive endpoint" from a "send endpoint" as these are now mutually exclusive.
+
+For example:
+
+** `send-config` now used by the `send-with-sync-mdn` and `send-with-async-mdn` operations
+** `listener-config` now used by the `as2-listener` source
+** `mdn-listener-config` now used by the `as2-mdn-listener` source
+
+
+== Requirements and Limitations
+
+There are some requirements and limitations you should be aware of before you upgrade to the latest version.
+
+=== Requirements
+
+Ensure you meet the following requirements before you migrate.
+
+Mule 4.1.0 or later
+
+Java 8
+
+=== Limitation
+
+Before you perform the migration, be aware of the following known limitation:
+
+* Signed copies of the document for nonrepudiation are not saved.
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade to version 5:
+
+. Download the AS2 version 5 connector from Exchange.
+
+. Delete the old namespace and add the new one.
+
+=== Before (AS2 v4 namespace)
+[source,xml,linenums]
+----
+
+```
+<mule  xmlns:as2="http://www.mulesoft.org/schema/mule/as2"
+    xmlns:doc="http://www.mulesoft.org/schema/mule/documentation"
+	xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="
+http://www.mulesoft.org/schema/mule/as2 http://www.mulesoft.org/schema/mule/as2-mule4/current/mule-as2.xsd
+http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
+```
+
+----
+
+=== After (AS2 v5 namespace)
+[source,xml,linenums]
+----
+```
+<mule xmlns:as2-mule4="http://www.mulesoft.org/schema/mule/as2-mule4"
+	xmlns="http://www.mulesoft.org/schema/mule/core"
+	xmlns:doc="http://www.mulesoft.org/schema/mule/documentation" xsi:schemaLocation="
+	http://www.mulesoft.org/schema/mule/as2-mule4 http://www.mulesoft.org/schema/mule/as2-mule4/current/mule-as2-mule4.xsd
+http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
+```
+----
+
```

Comment:
The following step came out of nowhere to me, because I had forgotten that all the previous text was actually one step!

Reply:
Hence, my rethinking of formatting....